File: core/src/com/colorflow/data/StorageInterface.java
package com.colorflow.data;

import java.util.List;

/**
 * Created by daniele on 28/07/17.
 */

public interface StorageInterface {
     int getCoins();
     String getVersion();
     int getRecord();
     List<String> getRings();
     List<String> getBonus();
     String getUsedRing();
     void incCoins(int coins);
     void setRecord(int score);
     void setVersion(String version);
     void setUsedRing(String ringId);
     void purchaseRing(int cost, String id);
     void purchaseBonus(int cost, String id);
}

File: core/src/com/colorflow/data/StorageInterface.kt
package com.colorflow.data

/**
 * Created by daniele on 28/07/17.
 */

interface StorageInterface {
    val coins: Int
    var version: String
    var record: Int
    val rings: List<String>
    val bonus: List<String>
    var usedRing: String
    fun incCoins(coins: Int)
    fun purchaseRing(cost: Int, id: String)
    fun purchaseBonus(cost: Int, id: String)
}
